golang/day_trader/src/main.go

```diff
@@ -94,19 +94,29 @@ var (
 )
 
 func (s *server) GetUser(ctx context.Context, req *pb.Command) (*pb.UserResponse, error) {
-	createUser(req.UserId)
-	return &pb.UserResponse{UserId: req.UserId}, nil
+	balance, stocks, err := userExists(req.UserId, req.Password)
+	var userResp *pb.UserResponse
+	if err == nil {
+		userResp = &pb.UserResponse{
+			UserId:  req.UserId,
+			Balance: balance,
+			Stocks:  stocks,
+		}
+	}
+	return userResp, err
 }
 
 func (s *server) CreateUser(ctx context.Context, req *pb.Command) (*pb.Response, error) {
-	createUser(req.UserId)
-	return &pb.Response{Message: "Created user with id" + req.UserId}, nil
+	return &pb.Response{UserId: req.UserId, Message: "Successfully created user"}, createUser(req.UserId, req.Password)
 }
 
 func (s *server) Add(ctx context.Context, req *pb.Command) (*pb.BalanceResponse, error) {
 	user := getUser(req.UserId)
-	user.updateUserBalance(ctx, req.Amount, true)
-	return &pb.BalanceResponse{UserId: req.UserId}, nil
+	user, err := user.updateUserBalance(ctx, req.Amount, true)
+	if err == nil {
+		err = errors.New(fmt.Sprintf("Amount: %f Balance: %f", req.Amount, user.Balance))
+	}
+	return &pb.BalanceResponse{UserId: user.Id, Balance: user.Balance}, err
```

Comment:
Maybe I am missing something but will this always return an error, one of which will contain non error information?

Reply:
Oh yeah lol, I was testing something here. I've removed this.